File: saizad-mvvm/src/main/java/com/saizad/mvvm/ActivityResult.java
package com.saizad.mvvm;

import android.app.Activity;

import androidx.annotation.Nullable;

public class ActivityResult<V> {

    private final int requestCode;
    private final int resultCode;
    @Nullable
    private V value;

    public ActivityResult(final int requestCode, final @Nullable V value) {
        this(requestCode, OK(), value);
    }

    public ActivityResult(final int requestCode, final int resultCode,
                          final @Nullable V value) {
        this.resultCode = resultCode;
        this.requestCode = requestCode;
        this.value = value;
    }

    public static int OK() {
        return Activity.RESULT_OK;
    }

    public boolean isCanceled() {
        return resultCode == Activity.RESULT_CANCELED;
    }

    public boolean isOk() {
        return resultCode == OK();
    }

    public boolean isRequestCode(final int requestCode) {
        return this.requestCode == requestCode;
    }

    @Nullable
    public V getValue() {
        return value;
    }
}

File: saizad-mvvm/src/main/java/com/saizad/mvvm/ActivityResult.kt
package com.saizad.mvvm

import android.app.Activity

class ActivityResult<V>(val requestCode: Int, val resultCode: Int, val value: V?) {

    constructor(requestCode: Int, value: V?) : this(requestCode, OK(), value) {}

    val isCanceled: Boolean
        get() = resultCode == Activity.RESULT_CANCELED

    val isOk: Boolean
        get() = resultCode == OK()

    fun isRequestCode(requestCode: Int): Boolean {
        return this.requestCode == requestCode
    }

    companion object {
        fun OK(): Int {
            return Activity.RESULT_OK
        }
    }

}